README-pl-pl.md

```diff
@@ -0,0 +1,1744 @@
+
+```js
+console.log("hello, world!"); // > hello, world!
+```
+
+**`//`** jest tylko komentarzem używanym w celu wyjaśnienia. Przykład:
+
+```js
+// Assigning a function to foo constant
+const foo = function() {};
+```
+
+# 👀 Przykłady
+
+## `[]` jest równe `![]`
+
+Tablica jest równa zanegowanej tablicy:
+
+```js
+[] == ![]; // -> true
+```
+
+### 💡 Wytłumaczenie:
+
+Abstrakcyjny operator równości przekształca obie strony na liczby, aby je porównać, a obie strony stają się liczbą `0` z różnych powodów. Tablice są prawdziwe, więc po prawej stronie przeciwieństwem prawdziwej wartości jest `false`, który jest następnie wymuszany na `0`. Po lewej jednak pusta tablica jest wymuszana na liczbę, nie będąc najpierw wartością logiczną, a puste tablice są wymuszane na `0`, mimo że są prawdziwe.
+
+Oto jak to wyrażenie upraszcza:
+
+```js
++[] == +![];
+0 == +false;
+0 == 0;
+true;
+```
+
+Zobacz też [`[]` is truthy, but not `true`](#-is-truthy-but-not-true).
+
+- [**12.5.9** Logical NOT Operator (`!`)](https://www.ecma-international.org/ecma-262/#sec-logical-not-operator)
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## `true` nie jest równe `![]`, ale też nie równe `[]`
+
+Tablica nie jest równa `true`, ale nie Tablica nie jest równa `true` również;
+Tablica jest równa `false`, nie Tablica jest równa `false` również:
+
+```js
+true == []; // -> false
+true == ![]; // -> false
+
+false == []; // -> true
+false == ![]; // -> true
+```
+
+### 💡 Wytłumaczenie:
+
+```js
+true == []; // -> false
+true == ![]; // -> false
+
+// According to the specification
+
+true == []; // -> false
+
+toNumber(true); // -> 1
+toNumber([]); // -> 0
+
+1 == 0; // -> false
+
+true == ![]; // -> false
+
+![]; // -> false
+
+true == false; // -> false
+```
+
+```js
+false == []; // -> true
+false == ![]; // -> true
+
+// According to the specification
+
+false == []; // -> true
+
+toNumber(false); // -> 0
+toNumber([]); // -> 0
+
+0 == 0; // -> true
+
+false == ![]; // -> true
+
+![]; // -> false
+
+false == false; // -> true
+```
+
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## prawda to fałsz
+
+```js
+!!"false" == !!"true"; // -> true
+!!"false" === !!"true"; // -> true
+```
+
+### 💡 Wytłumaczenie:
+
+Rozważ to krok po kroku:
+
+```js
+// true is 'truthy' and represented by value 1 (number), 'true' in string form is NaN.
+true == "true"; // -> false
+false == "false"; // -> false
+
+// 'false' is not the empty string, so it's a truthy value
+!!"false"; // -> true
+!!"true"; // -> true
+```
+
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## baNaNa
+
+```js
+"b" + "a" + +"a" + "a"; // -> 'baNaNa'
+```
+
+This is an old-school joke in JavaScript, but remastered. Here's the original one:
```

Comment:
To stary żart w JavaScript, ale odnowiony. Oto oryginał:

Reply:
Agree, thanks again.